x-pack/solutions/security/plugins/security_solution/public/siem_migrations/rules/components/migration_status_panels/migration_name.tsx

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { useState, useCallback } from 'react';
+import { EuiFlexGroup, EuiFlexItem, EuiInlineEditText } from '@elastic/eui';
+import type { MigrationReadyPanelProps } from './migration_ready_panel';
+import { useUpdateMigrationName } from '../../logic/use_update_migration_name';
+
+const MigrationName = React.memo<MigrationReadyPanelProps>(({ migrationStats }) => {
+  const [inlineEditValue, setInlineEditValue] = useState<string>(migrationStats.name);
+
+  const { mutate: updateMigrationName, isLoading: isUpdatingMigrationName } =
+    useUpdateMigrationName();
+  const handleInlineEditChange = useCallback((event: React.ChangeEvent<HTMLInputElement>) => {
+    setInlineEditValue(event.target.value);
+  }, []);
+
+  const handleEditCancel = useCallback(() => {
+    setInlineEditValue('');
+  }, []);
+
+  const handleUpdateMigrationName = useCallback(
+    (value: string) => {
+      updateMigrationName({ migrationId: migrationStats.id, name: value });
+    },
+    [updateMigrationName, migrationStats.id]
+  );
+
+  return (
+    <>
+      <EuiFlexItem grow={false}>
```

Comment:
I've noticed the collapsible panel expands and collapses on every click in the name, edit button, confirm, cancel... Could we prevent that by adding a click event handler here to stop the propagation?

```suggestion

  const stopPropagation = useCallback((e: React.MouseEvent) => {
    e.stopPropagation(); // prevent click events from bubbling up and toggle the collapsible panel
  }, []);

  return (
    <EuiFlexItem grow={false} onClick={stopPropagation}>
```